zerver/tests/test_realm.py

```diff
@@ -448,6 +448,47 @@ def test_user_group_edit_policy(self) -> None:
         result = self.client_patch('/json/realm', req)
         self.assert_json_error(result, 'Invalid user_group_edit_policy')
 
+    def test_invalid_integer_attribute_values(self) -> None:
+
+        integer_values = [key for key, value in Realm.property_types.items() if value is int]
+
+        invalid_values = dict(
+            bot_creation_policy=10,
+            create_stream_policy=10,
+            invite_to_stream_policy=10,
+            email_address_visibility=10,
+            message_retention_days=10,
+            video_chat_provider=0,
+            waiting_period_threshold=-10,
+            digest_weekday=10,
+            user_group_edit_policy=10,
+        )
+
+        # We need an admin user.
+        email = 'iago@zulip.com'
+        self.login(email)
+
+        for name in integer_values:
+            invalid_value = invalid_values.get(name)
+            if invalid_value is None:
+                raise AssertionError('No test created for %s' % (name,))
+
+            self.do_test_invalid_integer_attribute_value(name, invalid_value)
+
+    def do_test_invalid_integer_attribute_value(self, val_name: str, invalid_val: int) -> None:
+
+        possible_messages = {
```

Comment:
This was the best way of automating these tests without changing messages around the code I could think

Reply:
Yeah, this is reasonable.  That said, I think we're fine with changing error messages for consistency.